eZ/Publish/Core/FieldType/Image/IO/Legacy.php

```diff
@@ -149,7 +150,13 @@ public function loadBinaryFile($binaryFileId)
     public function loadBinaryFileByUri($binaryFileUri)
     {
         try {
-            return $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+            $image = $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+
+            if ($image instanceof MissingBinaryFile) {
+                $image = $this->draftIOService->loadBinaryFileByUri($binaryFileUri);
+            }
+
+            return $image;
         } catch (InvalidArgumentException $prefixException) {
             // InvalidArgumentException means that the prefix didn't match, NotFound can pass through
             try {
```

Comment:
@tomaszmadeyski We are not throwing InvalidArgumentException anywhere ourselves anymore? is that what caused this issue in to appear? If so maybe do a followup PR with option 1 for master _(or maybe 7.0 if we consider this break)_ where the throw doc block is removed. edyt @glye ?